gapis/api/vulkan/extensions/ext_host_query_reset.api

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2019 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Based off of the original vulkan.h header file which has the following
+// license.
+
+// Copyright (c) 2015 The Khronos Group Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and/or associated documentation files (the
+// "Materials"), to deal in the Materials without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Materials, and to
+// permit persons to whom the Materials are furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Materials.
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+///////////////
+// Constants //
+///////////////
+
+@extension("VK_EXT_host_query_reset") define VK_EXT_HOST_QUERY_RESET_SPEC_VERSION   1
+@extension("VK_EXT_host_query_reset") define VK_EXT_HOST_QUERY_RESET_EXTENSION_NAME "VK_EXT_host_query_reset"
+
+@extension("VK_EXT_host_query_reset")
+class VkPhysicalDeviceHostQueryResetFeaturesEXT {
+    VkStructureType    sType
+    void*              pNext
+    VkBool32           hostQueryReset
+}
+
+@extension("VK_EXT_host_query_reset")
+@indirect("VkDevice")
+cmd void vkResetQueryPoolEXT(
+    VkDevice           device,
+    VkQueryPool        queryPool,
+    u32                firstQuery,
+    u32                queryCount) {
+  if !(queryPool in QueryPools) { vkErrorInvalidQueryPool(queryPool) }
+  pool := QueryPools[queryPool]
+  if pool != null {
+    for i in (0 .. queryCount) {
+    query := firstQuery + i
+    if !(query < queryCount) { vkErrorQueryOutOfRange(queryPool, query) }
+    pool.Status[firstQuery + i] = QUERY_STATUS_INACTIVE
+    }
```

Comment:
indentation inside for loop, and reuse of `query` variable, and you probably want to compare to `len(pool.Status)` ?

```
    for i in (0 .. queryCount) {
￼      query := firstQuery + i
￼      if !(query < len(pool.Status)) {
         vkErrorQueryOutOfRange(queryPool, query)
       } else {
￼        pool.Status[query] = QUERY_STATUS_INACTIVE
       }
￼    }
```